cmake/bazel_to_cmake.py

```diff
@@ -0,0 +1,324 @@
+#!/usr/bin/env python3
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This is yet another bazel-to-cmake converter. It's independently written from
+scratch but relies on the same basic idea as others (including IREE's), namely:
+to let the python interpreter do the bulk of the work, exploiting the fact that
+both Bazel's BUILD syntax and Starlark (".bzl") languages are more or less
+subsets of Python.
+
+The main features that this converter supports and that others don't, justifying
+its existence as of early 2021, are:
+  1. Ad-hoc support for select(), generating CMake if()...elseif()... chains
+     parsing the condition keys (e.g. anything ending in ":windows" is interpreted as the condition "the target platform
+     is Windows"). This allows to just ignore config_setting, as we only care
+     about the config_setting names, not their actual implementation, as well
+     as all the variants from the Bazel 'selects' library.
+  2. Support for load(), loading macros from Starlark files.
+  3. Generates the entire project, including the project-root CMakeLists.txt.
+     The only non-generated files are cmake/*.cmake.
+"""
+
+import re
+import os
+import os.path
+import pickle
+import sys
+import datetime
+import itertools
+
+# Ruy's dependencies.
+external_targets = ['gtest', 'gtest_main', 'cpuinfo']
+
+# Text replacements [oldstring, newstring] pairs, applied on all BUILD and
+# Starlark files that we load. Only used by preprocess_input_text.
+replacements = [
+    ['$(STACK_FRAME_UNLIMITED)', ''],
+    ['native.cc_', 'cc_'],
+    ['selects.config_setting_group', 'config_setting_group'],
+    ['@com_google_googletest//:gtest', 'gtest'],
+    ['@com_google_googletest//:gtest_main', 'gtest_main'],
+    ['@cpuinfo//:cpuinfo_with_unstripped_include_path', 'cpuinfo'],
```

Comment:
In IREE we do a separate target mapping. Might be cleaner?

Reply:
Cleaner things are going to wait a bit - Ruy is not as active as IREE as a project so there is less motivation to be very clean.